tests/testtaglist.cpp

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: GPL-2.0
+#include "testtaglist.h"
```

Comment:
i would have moved the test into a separate commit, but i guess this is fine too. see if you'd prefer that?
the first commit message would need an amend in that case.